src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/AsyncTaskMethodBuilderTests.cs

```diff
@@ -13,212 +13,206 @@ namespace System.Threading.Tasks.Tests
-                Assert.True(t.Exception.InnerException is InvalidCastException, "    > FAILURE. Wrong exception found in builder (ATMB, build then fault)");
-                Assert.Throws<InvalidOperationException>(
-                   () => { atmb.SetResult(); });
-                Assert.Throws<InvalidOperationException>(
-                   () => { atmb.SetException(new Exception()); });
-            }
+            // Creating a task builder, building it, completing it successfully
+            var atmb = AsyncTaskMethodBuilder<int>.Create();
+            var t = atmb.Task;
+            Assert.Equal(TaskStatus.WaitingForActivation, t.Status);
+            atmb.SetResult(43);
+            Assert.Equal(TaskStatus.RanToCompletion, t.Status);
+            Assert.Equal(43, t.Result);
+        }
 
-            // Creating a task builder, completing it faulted, building it, and making sure it can't be reset
-            {
-                var atmb = AsyncTaskMethodBuilder.Create();
-                atmb.SetException(new InvalidCastException());
-                var t = atmb.Task;
-                Assert.True(t.Status == TaskStatus.Faulted, "    > FAILURE. Builder should be faulted after an exception occurs (ATMB, fault then build)");
-                Assert.True(t.Exception.InnerException is InvalidCastException, "    > FAILURE. Wrong exception found in builder (ATMB, fault then build)");
-                Assert.Throws<InvalidOperationException>(
-                   () => { atmb.SetResult(); });
-                Assert.Throws<InvalidOperationException>(
-                   () => { atmb.SetException(new Exception()); });
-            }
+        [Fact]
+        public static void TaskMethodBuilderT_DoesNotTouchSyncContext()
+        {
+            // Verify that AsyncTaskMethodBuilder<T> is not touching sync context
+            SynchronizationContext previousContext = SynchronizationContext.Current;
+            var trackedContext = new TrackOperationsSynchronizationContext();
+            SynchronizationContext.SetSynchronizationContext(trackedContext);
 
-            // Test cancellation
+            var atmb = AsyncTaskMethodBuilder<string>.Create();
+            Assert.Equal(0, trackedContext.TrackedCount);
+            atmb.SetResult("async");
+            Assert.Equal(0, trackedContext.TrackedCount);
+
+            SynchronizationContext.SetSynchronizationContext(previousContext);
+        }
+
+        // Incorrect usage for AsyncTaskMethodBuilder
+        [Fact]
+        [ActiveIssue("Hangs")]
+        public static void TaskMethodBuilder_IncorrectUsage()
+        {
+            var atmb = new AsyncTaskMethodBuilder();
+            Assert.Throws<ArgumentNullException>(() => { atmb.SetException(null); });
+        }
+
+        // Incorrect usage for AsyncVoidMethodBuilder
+        [Fact]
+        [ActiveIssue("Hangs")]
```

Comment:
This test is invalid as written; it makes sense that it hangs.  The void method builder posts an OperationStarted to the current SynchronizationContext if there is one, the point of which is to allow a host to wait for the corresponding OperationCompleted event.  xunit has a custom SynchronizationContext it uses, and it supports waiting until all of the operations have completed. You'll want to add a line after the assert that completes the atmb successfully so that the appropriate event is raised.  Alternatively you could null out the sync ctx prior to Create'ing the ATMB, and then reset the context in a finally block, as is done in some other tests.